torch/fx/experimental/proxy_tensor.py

```diff
@@ -1435,6 +1452,24 @@ def __torch_function__(
             if func is torch._C._set_grad_enabled:
                 func(*args, **kwargs)
             return node
+
+        if func in [
+            torch._functorch.vmap._add_batch_dim_in_graph,
+            torch._functorch.vmap._remove_batch_dim_in_graph,
+            torch._functorch.vmap._vmap_increment_nesting_in_graph,
+            torch._functorch.vmap._vmap_decrement_nesting_in_graph,
+            torch._functorch.vmap.lazy_load_decompositions,
+        ]:
+            _, proxies, _ = _fetch_proxies_and_all_constant_flag(args, self.tracer)
+            out_proxy = self.tracer.create_proxy(
+                "call_function",
+                func,
+                proxies,
+                {},
+            )
+            res = func(*args, **kwargs)
+            track_tensor_tree(res, out_proxy, constant=None, tracer=self.tracer)
+            return res
```

Comment:
why do these need to be special cased?

Reply:
Because unlike other func's, these functions require non-trivial handling of input and output proxies. By defualt, torch_function handler receives plain objects, not the proxies.